lib/matplotlib/axis.py

```diff
@@ -1077,10 +1077,15 @@ def _get_tick_bboxes(self, ticks, renderer):
                 [tick.label2.get_window_extent(renderer)
                  for tick in ticks if tick.label2.get_visible()])
 
-    def get_tightbbox(self, renderer):
+    def get_tightbbox(self, renderer, *, ignore_label=None):
         """
         Return a bounding box that encloses the axis. It only accounts
         tick labels, axis label, and offsetText.
+
+        If ``ignore_label`` is 'x', then the width of the label is collapsed
+        to near zero.  If 'y', then the height is collapsed to near zero. This
+        is for tight/constrained_layout to be able to ignore too-long labels
+        when doing their layout.  
```

Comment:
Perhaps instead use the same kwarg as for Axes.get_tightbbox (to simplify documentation), i.e. for_layout_only, and then lookup self.axis_name to know which direction to ignore?

Reply:
the docs are pretty sparse for `axis.get_tightlayout` and I wasn't sure about introspection on the axis name as a very robust way to do this.

Reply:
We look up rcparams based on axis_name, so that should be safe.

... by the way, how does that interact with polar plots?

Reply:
Not sure about polar plots....  Does tight_layout work with those at all?

Reply:
Looks so (e.g. try `subplots(2, 2, subplot_kw=dict(projection="polar")); title("foo"); tight_layout()`).

Reply:
OK, well that code runs fine with this PR....